crates/rust-analyzer/src/main_loop.rs

```diff
@@ -330,6 +330,12 @@ impl GlobalState {
                 .collect::<Vec<_>>();
 
             self.update_file_notifications_on_threadpool(subscriptions);
+
+            // Refresh semantic tokens if the client supports it.
+            if self.config.semantic_tokens_refresh {
+                self.semantic_tokens_cache.lock().clear();
+                self.send_request::<lsp_types::request::SemanticTokensRefesh>((), |_, _| ());
+            }
```

Comment:
I think this is the right spot to send a refresh request but am not 100% sure.

Reply:
Yeah, I think this is correct.